src/components/topo/ucc_sbgp.c

```diff
@@ -386,6 +456,9 @@ ucc_status_t ucc_sbgp_create(ucc_topo_t *topo, ucc_sbgp_type_t type)
     case UCC_SBGP_FULL:
         status = sbgp_create_full(topo, sbgp);
         break;
+    case UCC_SBGP_FULL_HOST_ORDERED:
+        status = sbgp_create_full_ordered(topo, sbgp);
```

Comment:
is it host list order preserved when doing this ordering?

Reply:
it sorts hosts by hashes, for instance
**original mapping**:
rank0 - hostA (hash: 0xB) socket 0
rank1 - hostA (hash: 0xB) socket 1
rank2 - hostA (hash: 0xB) socket 0
rank3 - hostA (hash: 0xB) socket 1
rank4 - hostB (hash: 0xA) socket 0
rank5 - hostB (hash: 0xA) socket 1
rank6 - hostB (hash: 0xA) socket 0
rank7 - hostB (hash: 0xA) socket 1
**reordering**:
rank4, rank6, rank5, rank7, rank0, rank2, rank1, rank3